src/Components/Users/UsersListTable.js

```diff
@@ -0,0 +1,276 @@
+    username: 'Bret',
+    email: 'Sincere@april.biz',
+  },
+  {
+    id: 2,
+    name: 'Ervin Howell',
+    username: 'Antonette',
+    email: 'Shanna@melissa.tv',
+  },
+  {
+    id: 3,
+    name: 'Clementine Bauch',
+    username: 'Samantha',
+    email: 'Nathan@yesenia.net',
+  },
+  {
+    id: 4,
+    name: 'Patricia Lebsack',
+    username: 'Karianne',
+    email: 'Karley_Dach@jasper.info',
+  },
+  {
+    id: 5,
+    name: 'Chelsey Dietrich',
+    username: 'Kamren',
+    email: 'Lucio_Hettinger@annie.ca',
+  },
+  {
+    id: 6,
+    name: 'Mrs. Dennis Schulist',
+    username: 'Leopoldo_Corkery',
+    email: 'Karley_Dach@jasper.info',
+  },
+  {
+    id: 7,
+    name: 'Kurtis Weissnat',
+    username: 'Elwyn.Skiles',
+    email: 'Telly.Hoeger@billy.biz',
+  },
+  {
+    id: 8,
+    name: 'Nicholas Runolfsdottir V',
+    username: 'Maxime_Nienow',
+    email: 'Sherwood@rosamond.me',
+  },
+  {
+    id: 9,
+    name: 'Glenna Reichert',
+    username: 'Delphine',
+    email: 'Chaim_McDermott@dana.io',
+  },
+  {
+    id: 10,
+    name: 'Clementina DuBuque',
+    username: 'Moriah.Stanton',
+    email: 'Rey.Padberg@karina.biz',
+  },
+  {
+    id: 11,
+    name: 'Patricia Reichert',
+    username: 'Karianne',
+    email: 'Julianne.OConner@kory.org',
+  },
+  {
+    id: 12,
+    name: 'Clementine Howell',
+    username: 'Samantha',
+    email: 'Sherwood@rosamond.me',
+  },
+  {
+    id: 13,
+    name: 'Chelsey Weissnat',
+    username: 'Kamren',
+    email: 'Chaim_McDermott@dana.io',
+  },
+];
+
+const UsersListTable = () => {
+  const [order, setOrder] = useState('asc');
+  const [orderBy, setOrderBy] = useState('name');
+  const [page, setPage] = useState(0);
+  const [usersList, setUsersList] = useState(mockedUsers);
+  const rowsPerPage = 10;
+
+  const descendingComparator = (a, b, orderBy) => {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+
+    return 0;
+  };
+
+  const getComparator = (order, orderBy) => {
+    return order === 'desc'
+      ? (a, b) => descendingComparator(a, b, orderBy)
+      : (a, b) => -descendingComparator(a, b, orderBy);
+  };
+
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === 'asc';
+
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const createSortHandler = (property) => (event) => {
+    handleRequestSort(event, property);
+  };
+
+  // Avoid a layout jump when reaching the last page with empty rows.
+  const emptyRows =
+    page > 0 ? Math.max(0, (1 + page) * rowsPerPage - usersList.length) : 0;
+
+  const deleteUser = (id) => {
+    const newUsersList = usersList.filter((user) => user.id !== id);
+    const total = usersList.length - 1;
+    const pages = total / rowsPerPage;
+
+    setUsersList(newUsersList);
+
+    if (page === pages && page !== 0) {
```

Comment:
What does this validation mean? Maybe saving its result in a properly named const should clarify what this is validating